packages/utils/src/error.ts

```diff
@@ -121,9 +121,16 @@ export function processError(err: any, diffOptions?: DiffOptions) {
   try {
     if (typeof err.message === 'string')
       err.message = normalizeErrorMessage(err.message)
+  }
+  catch {}
 
-    if (typeof err.cause === 'object' && typeof err.cause.message === 'string')
-      err.cause.message = normalizeErrorMessage(err.cause.message)
+  // some Error implementations may not allow rewriting cause
+  // in most cases, the assignment will lead to "err.cause = err.cause"
+  try {
+    if (!seen.has(err) && typeof err.cause === 'object') {
+      seen.add(err)
```

Comment:
@sheremet-va - I just added a check to avoid infinitely loping if cause is self-referencing itself later. Plus created a dedicated test for it